src/controller/favorites.js

```diff
@@ -0,0 +1,93 @@
+const router = require("express").Router(),
+	unauthError = require("../responses/unauthenticated.json"),
+	actions = {
+		findAll: require("../actions/favorites/all"),
+		add: {
+			album: require("../actions/favorites/add/album"),
+			artist: require("../actions/favorites/add/artist"),
+			genre: require("../actions/favorites/add/genre"),
+			song: require("../actions/favorites/add/song")
+		},
+		remove: {
+			album: require("../actions/favorites/remove/album"),
+			artist: require("../actions/favorites/remove/artist"),
+			genre: require("../actions/favorites/remove/genre"),
+			song: require("../actions/favorites/remove/song")
+		}
+	};
+
+module.exports = router
```

Comment:
I wonder if this could be solved with a payload instead of having to duplicate code like this. Something along the lines of:
```
	.post('/', (req, res) => {
		if (!req.authenticated) {
			return res.json(unauthError);
		}

		const favorables = [
			"genre",
			"artist",
			"album",
			"song",
		];

		if (!favorables.includes(req.body.type)) {
			return {
				ok: false,
				message: "Yo what the fuck"
			};
		}

		// something like this, not tested!
		actions.add[req.body.type]()(req.token.userId, req.body.genreId).then(result => {
			return res.json(result);
		});
	})
```